### PR TITLE
Help button fix

### PR DESCRIPTION
**Pull request by Mahmud-Khan**
### Summary

- Fixed the help-button alignment issue with. The help button would stick to the bottom if the content does not fill the screen. But if the content height is more than viewport height the help-button would be below the content.
help_button_fix_static

Please add your name to the [CONTRIBUTORS.md] file. Adding your name to the [CONTRIBUTORS.md] file signifies agreement to all rights and reservations provided by the [License].

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: packages/terra-consumer-layout/src/Layout.scss [hunk 1]
```diff
@@ -55,8 +55,12 @@
   }
 
   .help-button {
-    margin: 15px;
+    padding: 20px;
+    position: fixed;
+    position: sticky;
+    right: 0;
     text-align: right;
+    top: 90%;
```
Comment (mhemesath) [inline packages/terra-consumer-layout/src/Layout.scss]: this will be variable depending on the height of the screen.... why not use bottom instead?

Comment (Mahmud-Khan) [reply packages/terra-consumer-layout/src/Layout.scss]: Right now the help-button is part of the main container. Used bottom keeping the button inside the main container. Since the sticky position is relative to the parent component the button moves to the top when there is no content like the following example.
screen shot 2017-09-06 at 4 48 59 pm

Bottom works in we move the help-button outside the main container. But in that case, the help button is not relative to the content of the main container.

Comment (mhemesath) [reply packages/terra-consumer-layout/src/Layout.scss]: help button should stay at the bottom of the page I think

File: packages/terra-consumer-layout/CHANGELOG.md [hunk 2]
```diff
@@ -4,3 +4,6 @@ ChangeLog
 # 0.1.0-BETA.1 - (September 06, 2017)
 -----------------
 Initial beta release for the layout component
+
+### Changed
+- Fixed the help-button alignment issue.
```
Comment (alex-bezek) [inline packages/terra-consumer-layout/CHANGELOG.md]: I released a new beta today, so i think this needs to be part of a new beta tag in the changelog

File: packages/terra-consumer-layout/src/Layout.scss [hunk 3]
```diff
@@ -55,8 +55,12 @@
   }
 
   .help-button {
-    margin: 15px;
+    padding: 20px;
+    position: fixed;
+    position: sticky;
```
Comment (alex-bezek) [inline packages/terra-consumer-layout/src/Layout.scss]: how are we defining postion twice? Sorry not super familar with sticky

Comment (Mahmud-Khan) [reply packages/terra-consumer-layout/src/Layout.scss]: Edge/ie does not support sticky position. I was trying to apply CSS specific to edge/ie and set the position fixed for them. Talked to Mike and he suggested to set position: fixed above sticky. Tested it and from my understanding, if sticky is not supported it falls back to fixed.

**Comment by KevinJJackson** [comment]
your main content is scrollable even when there is no content, might need to change the min-height on that a bit.

File: packages/terra-consumer-layout/src/Layout.scss [hunk 4]
```diff
@@ -57,11 +63,13 @@
 
   .help-button {
     margin: 15px;
+    position: static;
```
Comment (mhemesath) [inline packages/terra-consumer-layout/src/Layout.scss]: this shouldn't be needed.

File: packages/terra-consumer-layout/src/Layout.scss [hunk 5]
```diff
@@ -57,11 +63,13 @@
...
     text-align: right;
 
     @media screen and (min-width: $terra-medium-breakpoint) {
       bottom: 15px;
       margin: 0;
+      padding: 0;
```
Comment (mhemesath) [inline packages/terra-consumer-layout/src/Layout.scss]: this shouldn't be needed if the component doesn't have padding outside of the media query

File: packages/terra-consumer-nav/src/components/modal/Modal.scss [hunk 6]
```diff
@@ -5,13 +5,15 @@
     background: var(--terra-consumer--modal-background, #fff);
     border-radius: 5px;
     left: 20px;
+    max-height: calc(100% - 80px);
```
Comment (mhemesath) [inline packages/terra-consumer-nav/src/components/modal/Modal.scss]: why is this needed?

File: packages/terra-consumer-layout/src/Layout.scss [hunk 7]
```diff
@@ -34,6 +34,12 @@
     white-space: break-word;
   }
 
+  .main-content {
+    @media screen and (max-width: $terra-medium-breakpoint) {
+      min-height: calc(100vh - 120px);
```
Comment (mhemesath) [inline packages/terra-consumer-layout/src/Layout.scss]: can you add a comment why we are doing this

File: packages/terra-consumer-layout/src/Layout.scss (hunk 5 above)
Comment (Mahmud-Khan) [reply packages/terra-consumer-layout/src/Layout.scss]: Will remove that. Used padding instead of margin before so had to set padding 0 for mobile, not needed now.

File: packages/terra-consumer-nav/src/components/modal/Modal.scss (hunk 6 above)
Comment (Mahmud-Khan) [reply packages/terra-consumer-nav/src/components/modal/Modal.scss]: Scroll bar does not appear if max-height is not set. The modal header(title and close button) height is 60px and space below modal content is 10px/20px for mobile/tablet. Subtracted the height of the header and space below content from screen height.

File: packages/terra-consumer-layout/src/Layout.scss (hunk 7 above)
Comment (Mahmud-Khan) [reply packages/terra-consumer-layout/src/Layout.scss]: The help button height is 40px and it has 20px margin top-bottom. Subtracted 80px initially from 100vh to stop the help button sticking on the bottom of the screen. However, even for empty content scroll appears like below
layout_scroll
Added some extra space to stop scroll when content empty.

Comment (mhemesath) [reply packages/terra-consumer-layout/src/Layout.scss]: I mean in the code :)

File: packages/terra-consumer-nav/src/components/modal/Modal.scss (hunk 6 above)
Comment (mhemesath) [reply packages/terra-consumer-nav/src/components/modal/Modal.scss]: I guess I don't understand why a help button fix needs to modify modal styles?

Comment (Mahmud-Khan) [reply packages/terra-consumer-nav/src/components/modal/Modal.scss]: Sorry, there was a modal content scrolling issue in Nav. I fixed it as part of the same PR. I can create a separate PR.
